### PR TITLE
Extract the edit-post module as a reusable package

### PR DESCRIPTION
**Pull request by youknowriad**
This PR extracts the edit-post package into the npm packages. The main reason is to make it usable in WordPress Core as a npm dependency. This package is not meant to be used outside the WordPress context though.

This is also the last Gutenberg package to move to the packages folder which means all of the Gutenberg JS code is a package now. There's only one deprecated component left at the root-level which will be able to remove in one or two releases.

### CODE REVIEW

File: packages/list-reusable-blocks/package.json [hunk 1]
```diff
@@ -1,14 +1,13 @@
 {
 	"name": "@wordpress/list-reusable-blocks",
-	"version": "1.0.1",
+	"version": "1.0.0-alpha.0",
```
Comment (gziolo) [inline packages/list-reusable-blocks/package.json]: Not sure if npm will allow downgrading

Comment (youknowriad) [reply packages/list-reusable-blocks/package.json]: It was not released :)

**Review by gziolo — APPROVED** [review]
I can confirm that everything still works as expected. I added a few commits so it would be nice to have another check for the code I added.

Amazing work @youknowriad on moving all remaining code to packages. This one marks another milestone. Looking forward to see the outcome of work started by @omarreiss and @atimmer to see it consumed in WordPress core.

**Comment by youknowriad** [comment]
Thanks for the help @gziolo

**Review by tofumatt — COMMENTED** [review]
> This package is not meant to be used outside the WordPress context though.

Can you still have `post-install` messages for `npm` packages? Might be nice if our Gutenberg package displayed some kind of message for our (currently) WordPress-specific packages. 🤷‍♂️

**Comment by youknowriad** [comment]
@tofumatt seems like a good idea, we'll have to make sure it's not "a big warning" as the message will show up in WordPress Core and people should feel confident ignoring it.

**Comment by gziolo** [comment]
Can we try to hide this message behind env flag so we could skip it in core?

**Pull request merged**